churaverse-plugins-server/src/churarenPlugin/churarenBossAttackPlugin/domain/bossAttackService.ts

```diff
@@ -0,0 +1,77 @@
+import { IBossAttackRepository } from './IBossAttackRepository'
+import { BossAttack } from './bossAttack'
+import { WorldMap } from '@churaverse/map-plugin-server/domain/worldMap'
+import { IMessageSender } from '@churaverse/network-plugin-server/interface/IMessageSender'
+import { IMainScene, Direction, IEventBus, Position, EntitySpawnEvent } from 'churaverse-engine-server'
+import { BossAttackSpawnMessage, BossAttackSpawnData } from '../message/bossAttackSpawnMessage'
+import { uniqueId } from '@churaverse/churaren-core-plugin-server/utils/uniqueId'
+
+/**
+ * 衝突した or 消滅時間に達したボス攻撃を削除
+ * @param onDelete 削除時に実行する関数.引数に削除されるボス攻撃のインスタンスとidを取る
+ */
+export function removeDieBossAttack(
+  bossAttacks: IBossAttackRepository,
+  onDelete: (bossAttackId: string) => void
+): void {
+  bossAttacks.getAllId().forEach((bossAttackId) => {
+    const bossAttack = bossAttacks.get(bossAttackId)
+    if ((bossAttack?.isDead ?? false) === true) {
+      bossAttacks.delete(bossAttackId)
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      onDelete(bossAttackId) // null合体でundefinedでないことは確定
+    }
+  })
+}
+
+/**
+ * BossAttackRepository内の全ボス攻撃を微小時間分だけ移動
+ * その際にワールド外に出たボス攻撃をdie
+ */
+export function moveBossAttacks(dt: number, bossAttacks: IBossAttackRepository, worldMap: WorldMap): void {
+  bossAttacks.getAllId().forEach((bossAttackId) => {
+    const bossAttack = bossAttacks.get(bossAttackId)
+    if (bossAttack !== undefined && bossAttack.isDead !== true) {
+      bossAttack.move(dt)
+      if (
+        bossAttack.position.x < 0 ||
+        bossAttack.position.x > worldMap.width ||
+        bossAttack.position.y < 0 ||
+        bossAttack.position.y > worldMap.height
+      ) {
+        bossAttack.die()
+      }
+      // 常に位置を更新（当たり判定システムに反映）
+      bossAttacks.updateActor(bossAttackId, bossAttack)
+    }
+  })
+}
+
+/**
+ * ボス攻撃生成をfrontendに通知する
+ */
+export function sendSpawnedBossAttack(
+  messageSender: IMessageSender<IMainScene>,
+  eventBus: IEventBus<IMainScene>,
+  startPos: Position,
+  bossId: string
+): void {
+  const gap = 100
+  if (startPos === undefined) return
+  // 四方向に攻撃を生成
+  Object.values(Direction).forEach((direction) => {
```

Comment:
変数は使用箇所にできるだけ近くなるように宣言した方が良いです。

この場合だと、`if (startPos === undefined) return`の処理にはgapは何も関係のない値ですが、if文の前で宣言されていることにより読み手はこの条件式に何らかの形で使われているものとして考えてしまいます。

```suggestion
  if (startPos === undefined) return
  // 四方向に攻撃を生成
  const gap = 100
  Object.values(Direction).forEach((direction) => {
```